### PR TITLE
First working code no errors or warnings steps 1 thru 9

### PR DESCRIPTION
**Pull request by dpsproservices**
Need to do more testing but here is the pull request.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -53,13 +55,53 @@ struct Wrapper
...
+        std::cout << "Wrapper::print(" << point.toString() << ")" << std::endl;
+    }
+
+private:
+    Point point;
+};
+
+// #3
+template<typename T, typename ... Args>
+void variadicHelper(T first, Args&& ... args)
```
Comment (matkatmusic) [inline main.cpp]: Your first param should be an R-value reference.  The current form is forcing a copy.  
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -53,13 +55,53 @@ struct Wrapper
...
+    Wrapper<T>(std::forward<T>(first)).print(); // #6
+    variadicHelper( std::forward<Args>(args) ... ); //recursive call
 }
 
+// #4
+template<typename T>
+void variadicHelper(T first)
+{
+    Wrapper<T>(std::forward<T>(first)).print(); // #6
+}
```
Comment (matkatmusic) [inline main.cpp]: Your param should be an R-value reference.  The current form is forcing a copy.  
- [ ] revise

File: type_name.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  suffix = "]";
+#elif defined(_MSC_VER)
+  name = __FUNCSIG__;
+  prefix = "auto __cdecl type_name<";
+  suffix = ">(void) noexcept";
+#endif
+  name.remove_prefix(prefix.size());
+  name.remove_suffix(suffix.size());
+  return name;
+}
```
Comment (matkatmusic) [inline type_name.h:37]: This is very clever!

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (dpsproservices) [reply main.cpp]: Done.

File: main.cpp (hunk 1 above)
Comment (dpsproservices) [reply main.cpp]: Done.

File: type_name.h (hunk 3 above)
Comment (dpsproservices) [reply type_name.h:37]: Simplified it as we are running replit with the argument -std=c++17, and  lines 22 through 24 are what assign the name, prefix and suffix. Removed lines 25 through 33 as they refer to macros which are meant for other C++ environments where __GNUC__ (GNU compilers that use the C preprocessor) or _MSC_VER  (MS Visual C++ compiler).

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**